### ISSUE TITLE
Publish milli in crates.io

### ISSUE THREAD

**Issue by hermeGarcia** [meilisearch/meilisearch#3367]
Hi! Sorry to bother, 
I am interested in using the milli engine inside a project.  Unfortunately the only way my project can afford to do so is by embedding the engine.
Could milli be published in crates.io?

**Comment by curquiza** [comment]
Here are some updates for this issue:

As we discussed a little bit [here](https://github.com/meilisearch/meilisearch/pull/3346), all our crates (milli included) will keep the same version as meilisearch since the crates are not published to [crates.io](http://crates.io/)

Once we will decide to publish milli, we will use a different version for milli compared to the other crates of the repo. However, milli will not be stable (by using `v0.X.Y` format and not `vX.Y.Z`). Indeed maintaining a stable milli crate would add a lot of burden to the team to improve our search engine in a smooth way.

Also, to be transparent, publishing milli is not a priority for the team for the moment (I mean at least in the coming months). It will bring additional (and probably complex) processes we don't need to provide a search engine with a good user experience :blush:
Meanwhile, you can still use milli by using the GitHub crate URL

**Comment by raphaelauv** [comment]
@curquiza is milli "enough" for using it as an embedded geospacial index database ? thanks

**Comment by GregoryConrad** [comment]
@raphaelauv Yes. You will miss out on meilisearch features like the autobatcher but it works just fine on its own and has the regular geo features from meilisearch.

**Comment by irevoire** [comment]
You will be missing the `_geoDistance`, but it's not hard to compute on your side. Here is our code to do it: https://github.com/meilisearch/meilisearch/blob/main/milli/src/lib.rs#L198-L205
Also, FYI, if you only need the geospatial index database, we’re using an `RTree` from the [`rstar`](https://docs.rs/rstar/latest/rstar/) crate for our database.

**Comment by Kinrany** [comment]
> It will bring additional (and probably complex) processes we don't need to provide a search engine with a good user experience

Counterpoint: being able to use the engine in-process contributes to developer experience, and a crates.io package is much better at this compared to compiling from source 🙂
